.github/workflows/release.yml

```diff
@@ -13,6 +13,7 @@ env:
   rust_version: 1.61.0
 
 name: Release smithy-rs
+run_name: ${{ github.workflow }} - ${{ inputs.dry_run && "Dry run" || "Production run" }}
```

Comment:
See [`run-name`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name).

`inputs.dry_run && "Dry run" || "Production run"` is a poor man's `inputs.dry_run ? "Dry run" : "Production run"`   (the latter is not supported in GH actions).